edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/KubernetesRuntimeInfoProvider.cs

```diff
@@ -110,31 +110,55 @@ public async Task<Stream> GetModuleLogs(string module, bool follow, Option<int>
                 cancellationToken: cancellationToken);
         }
 
-        public Task<SystemInfo> GetSystemInfo()
+        public async Task<SystemInfo> GetSystemInfo()
         {
-            // Comment this out to unblock testing.
-            // V1NodeList k8SNodes = await this.client.ListNodeAsync();
-            // string osType = string.Empty;
-            // string arch = string.Empty;
-            // string version = string.Empty;
-            // if (k8SNodes.Items != null)
-            // {
-            //     V1Node firstNode = k8SNodes.Items.FirstOrDefault();
-            //     if (firstNode?.Status?.NodeInfo != null)
-            //     {
-            //         osType = firstNode.Status.NodeInfo.OperatingSystem;
-            //         arch = firstNode.Status.NodeInfo.Architecture;
-            //         version = firstNode.Status.NodeInfo.OsImage;
-            //     }
-            //     else
-            //     {
-            //         Events.NullNodeInfoResponse(firstNode?.Metadata?.Name ?? "UNKNOWN");
-            //     }
-            // }
+            V1NodeList k8sNodes = await this.client.ListNodeAsync();
+            string accumulatedArch = string.Empty;
+
+            // Use "Kubernetes" and the current API version as analogy to device OS/version
             string osType = "Kubernetes";
-            string arch = "Kubernetes";
-            string version = "Kubernetes";
-            return Task.FromResult(new SystemInfo(osType, arch, version));
+            string version = k8sNodes.ApiVersion;
+
+            // Enumerate each node and count the unique architectures
+            var archList = new List<(string arch, int count)>();
```

Comment:
I think it's a very C/C++ way of doing this stuff. In a C# we have a very powerful way to work with collections called LINQ. 
This code can be re-written in a more declarative way:
```
		var achitectures = k8sNodes.Items
			.Where(node => node?.Status?.NodeInfo != null)
			.Select(node => node.Status.Status.NodeInfo.Architecture)
			.GroupBy(architecture => architecture)
			.Select(architectures => $"{architectures.Key} ({architectures.Count})";
		
		var arch = string.Join("\t", achitectures);
		return new SystemInfo(osType, arch, version);
```